sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -453,12 +467,12 @@ class AstBuilder extends DataTypeAstBuilder
       case table: InsertIntoTableContext =>
         val (relationCtx, options, cols, partition, ifPartitionNotExists, byName)
         = visitInsertIntoTable(table)
-        withIdentClause(relationCtx, ident => {
+        withIdentClause(relationCtx, Seq(query), (ident, otherPlans) => {
           val insertIntoStatement = InsertIntoStatement(
             createUnresolvedRelation(relationCtx, ident, options),
             partition,
             cols,
-            query,
+            otherPlans.head,
```

Comment:
Are you sure that everywhere you call `.head`, it won't fail with `NoSuchElementException`. Could you clarify your assumption, please.

Reply:
For all the places where we call `.head` we have otherPlans that was defined as `Seq(query)`, so it will have exactly one element. But Seq was used, if in the future the need for multiple plans is needed.